stl/inc/filesystem

```diff
@@ -3588,8 +3588,16 @@ namespace filesystem {
     inline bool remove(const path& _Target) {
         // remove file _Target (even if it is a directory); returns whether the file was removed
         // note !exists(_Target) is not an error, and merely returns false
-        const auto _Result = __std_fs_remove(_Target.c_str());
+        auto _Result = __std_fs_remove(_Target.c_str());
         if (_Result._Error != __std_win_error::_Success) {
+            if (_Result._Error == __std_win_error::_Access_denied) {
+                // Try to remove the ReadOnly Attribute and try again
+                const auto _Perm_change_err = __std_fs_change_permissions(_Target.c_str(), false, false);
+                if (_Perm_change_err != __std_win_error::_Success) {
+                    _Throw_fs_error("remove", _Perm_change_err, _Target);

```

Comment:
Previously, the exception-throwing and `error_code` overloads had repeated implementations (instead of one being implemented in terms of the other) because those implementations were small. Now that significant additional logic is needed to handle read-only files, we should centralize the logic. I recommend making the `error_code` implementation the fundamental one, and then moving the exception-throwing implementation below, so the exception-throwing overload can call the `error_code` overload.

Example above:
https://github.com/microsoft/STL/blob/54cce5e15e295181e1587d2b4e2e5df50faa0702/stl/inc/filesystem#L3523-L3532

Reply:
Have made the suggested changes